### PR TITLE
Fetch all movies from database

### LINKED ISSUE #26: GET all movies from API

### LINKED ISSUE #38: create API calls file?

### PR DESCRIPTION
**Pull request by clairefields15**
# Description
Fetch all movies from API! Added a console.error which will need to become an error message. Our loading message isn't showing up currently, but maybe it's hidden behind the header or something... adding an error component should fix this issue.

Fixes # (issue)
closes #38 
closes #26

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [x] Console Dev Tools

### CODE REVIEW

File: src/apiCalls/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+const baseURL = 'https://rancid-tomatillos.herokuapp.com/api/v2/';
+
+export async function fetchMovies(endpoint) {
+  const response = await fetch(`${baseURL}${endpoint}`);
+  return checkForErrors(response);
+}
+
+function checkForErrors(response) {
+  if (!response.ok) {
+    throw new Error(response.status + ' ' + response.statusText);
```
Comment (novaraptur) [inline src/apiCalls/apiCalls.js:11]: For later: implement Error component?

File: src/components/App/App.js [hunk 2]
```diff
@@ -4,41 +4,51 @@ import './App.css';
...
-      console.log("Scroll down to movie cards");
+      console.log('Scroll down to movie cards');
     } else {
       this.updateSelectedMovie(null);
     }
+  };
+
+  componentDidMount() {
+    fetchMovies('movies')
+      .then(data => this.setState({ movies: data.movies }))
```
Comment (novaraptur) [inline src/components/App/App.js:34]: Glad this reduces the length of tapping into our movies, too

File: src/components/Movies/Movies.js [hunk 3]
```diff
@@ -11,13 +11,13 @@ class Movies extends Component {
 
   filterMovies = id => {
     const { movies } = this.props;
-    const clickedMovie = movies.movies.find(movie => id === movie.id);
+    const clickedMovie = movies.find(movie => id === movie.id);
```
Comment (novaraptur) [inline src/components/Movies/Movies.js:14]: A single 'movies' is way easier than movies.movies!

**Review by novaraptur — APPROVED** [review]
Looks good! I'll get to work on fetching a single movie.

File: src/apiCalls/apiCalls.js (hunk 1 above)
Comment (kaylaewood) [reply src/apiCalls/apiCalls.js:11]: Great work pulling these functions into a separate file!

File: src/components/App/App.js [hunk 4]
```diff
@@ -4,41 +4,51 @@ import './App.css';
...
+import util from '../../apiCalls/util';
 
 class App extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      movies: movieData,
-      movie: singleMovie,
+      movies: [],
       selectedMovie: null
```
Comment (kaylaewood) [inline src/components/App/App.js:15]: I like these names better than `singleMovie` and `movieData`!

File: src/components/App/App.js [hunk 5]
```diff
@@ -4,41 +4,51 @@ import './App.css';
...
+      console.log('Scroll down to movie cards');
     } else {
       this.updateSelectedMovie(null);
     }
+  };
+
+  componentDidMount() {
+    fetchMovies('movies')
+      .then(data => this.setState({ movies: data.movies }))
+      .catch(err => console.error(err));
```
Comment (kaylaewood) [inline src/components/App/App.js:35]: How will the user know if something went wrong?

File: src/components/FeatMovie/FeatMovie.js [hunk 6]
```diff
@@ -1,17 +1,21 @@
 import React from 'react';
 import './FeatMovie.css';
 
-const FeatMovie = ({props}) => {
-  const movieIndex = Math.floor(Math.random() * props.movies.movies.length);
+const FeatMovie = ({ props }) => {
+  const movieIndex = Math.floor(Math.random() * props.movies.length);
```
Comment (kaylaewood) [inline src/components/FeatMovie/FeatMovie.js:5]: `index` can be a confusing term to use in this context, as we often think of index as the location of an element in an array. I might consider using `movieId` or `movieKey` instead.

File: src/components/Movies/Movies.js (hunk 3 above)
Comment (kaylaewood) [inline src/components/Movies/Movies.js:14]: Great work reorganizing the data so you avoid `movies.movies`

**Review by kaylaewood — COMMENTED** [review]
Overall, great work! No need to make any of the changes I recommended, but feel free to read through my comments and implement any changes you see fit. Let me know if you have any questions!

**Pull request merged**